### PR TITLE
Consider Null inputs

### PR DESCRIPTION
**Pull request by bernt-matthias**
as a possibility to allow the pick value tool to create an empty file.

Allows to connect an optional dataset to the pick value tool .. 

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: lib/galaxy/tools/__init__.py [hunk 1]
```diff
@@ -2857,6 +2857,8 @@ def exec_after_process(self, app, inp_data, out_data, param_dict, job=None, **kw
             if val.output_type == "data":
                 with open(out_data[key].get_file_name()) as f:
                     src = json.load(f)
+                if src == None:
```
Comment (mvdbeek) [inline lib/galaxy/tools/__init__.py]:
```suggestion
                if src is None:
```

**Comment by bernt-matthias** [comment]
I tested this with the following workflow:

Screenshot from 2024-02-15 16-10-56

[Galaxy-Workflow-Unnamed_Workflow.ga.txt](https://github.com/galaxyproject/galaxy/files/14298588/Galaxy-Workflow-Unnamed_Workflow.ga.txt)

**Review by mvdbeek — APPROVED** [review]
I've taken your workflow and used that as a test. Thanks for working on this!

**Comment by mvdbeek** [comment]
We should really add all these things as FAQ or mini tutorial to https://training.galaxyproject.org/training-material/topics/galaxy-interface/ ... there's so many cool things you can do that we never documented 😅

**Comment by bernt-matthias** [comment]
> We should really add all these things as FAQ or mini tutorial to https://training.galaxyproject.org/training-material/topics/galaxy-interface/ ... there's so many cool things you can do that we never documented 😅

Sounds like a good idea. We could start with an issue to collect ideas and start once we have 2 or 3. Once there is the faq / tutorial it's easier to add in the future.

**Comment by bernt-matthias** [comment]
https://github.com/galaxyproject/training-material/issues/4738

**Pull request merged**